src/content/Institutions/InstitutionView.tsx

```diff
@@ -1,7 +1,322 @@
+  const { lastSearchParams } = useSearchParamsContext();
+  const {
+    handleSubmit,
+    register,
+    control,
+    reset,
+    formState: { errors },
+  } = useForm<FormInput>({
+    mode: "onSubmit",
+    reValidateMode: "onSubmit",
+    defaultValues: {
+      name: "",
+      status: "Active",
+    },
+  });
+
+  const [saving, setSaving] = useState<boolean>(false);
+
+  const manageInstitutionsPageUrl = `/institutions${lastSearchParams?.["/institutions"] ?? ""}`;
+
+  const { loading: retrievingInstitution } = useQuery<GetInstitutionResp, GetInstitutionInput>(
+    GET_INSTITUTION,
+    {
+      variables: { _id },
+      skip: isNew,
+      onCompleted: (data) => {
+        const { name, status } = data?.getInstitution || {};
+        reset({ name, status });
+      },
+      context: { clientName: "backend" },
+      fetchPolicy: "no-cache",
+      onError: (error) => onError([error], "Unable to retrieve Institution."),
+    }
+  );
+
+  const [createInstitution] = useMutation<CreateInstitutionResp, CreateInstitutionInput>(
+    CREATE_INSTITUION,
+    {
+      context: { clientName: "backend" },
+      fetchPolicy: "no-cache",
+      errorPolicy: "all",
+      onError: (error) => {
+        if (error.networkError) {
+          onError([error], "Unable to create a new Institution.");
+        }
+      },
+    }
+  );
+
+  const [updateInstitution] = useMutation<UpdateInstitutionResp, UpdateInstitutionInput>(
+    UPDATE_INSTITUTION,
+    {
+      context: { clientName: "backend" },
+      fetchPolicy: "no-cache",
+      errorPolicy: "all",
+      onError: (error) => {
+        if (error.networkError) {
+          onError([error], "Unable to save changes.");
+        }
+      },
+    }
+  );
+
+  const onError = <T extends { message: string }>(errors: readonly T[], message: string) => {
+    if (!errors?.length) {
+      return;
+    }
+
+    const flatErrors = errors?.flatMap((err) => err.message)?.join(" ");
+    Logger.error(`InstitutionView: ${message} ${flatErrors}`);
+    enqueueSnackbar(flatErrors, { variant: "error" });
+  };
+
+  const handleCreateInstitution = async ({ name, status }: FormInput) => {
+    const { data: d, errors } = await createInstitution({ variables: { name, status } });
+
+    if (errors || !d?.createInstitution?._id) {
+      onError(errors, "Unable to create a new Institution.");
+      return;
+    }
+
+    enqueueSnackbar("Institution added successfully.", { variant: "success" });
+  };
+
+  const handleUpdateInstitution = async ({ name, status }: FormInput) => {
+    const { data: d, errors } = await updateInstitution({ variables: { _id, name, status } });
+
+    if (errors || !d?.updateInstitution?._id) {
+      onError(errors, "Unable to save changes.");
+      return;
+    }
+
+    enqueueSnackbar("Institution updated successfully.", { variant: "success" });
+    reset({ ...d.updateInstitution });
+  };
+
+  const onSubmit = async (data: FormInput) => {
```

Comment:
Trivial suggestion, but could we just add the RHF submit handler typing here?

```suggestion
  const onSubmit: SubmitHandler<InputForm> = async (data: FormInput) => {
```

Reply:
Thanks for the suggestion! Added in [3825bf5](https://github.com/CBIIT/crdc-datahub-ui/pull/682/commits/3825bf5ce5160292bf3435f93853195135657237).